integrations/cdi/common-cdi/configurable/src/main/java/io/helidon/integrations/cdi/configurable/ConfiguredProperties.java

```diff
@@ -0,0 +1,261 @@
+/*
+ * Copyright (c) 2024 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.helidon.integrations.cdi.configurable;
+
+import java.io.InputStream;
+import java.io.Reader;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.VarHandle;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+import java.util.function.BiConsumer;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+
+import org.eclipse.microprofile.config.Config;
+
+/**
+ * A lazy read-only {@link Properties} implementation backed by a {@link Config} instance.
+ */
+final class ConfiguredProperties extends Properties {
+
+    private static final VarHandle ENTRIES;
+
+    static {
+        try {
+            ENTRIES = MethodHandles.lookup().findVarHandle(ConfiguredProperties.class, "entries", Map.class);
+        } catch (ReflectiveOperationException e) {
+            throw new ExceptionInInitializerError(e);
+        }
+    }
+
+    @SuppressWarnings("unused")
+    private volatile Map<Object, Object> entries;
+    private final Map<Object, String> keys = new HashMap<>();
+    private final Config config;
+
+    ConfiguredProperties(Config config) {
```

Comment:
You can remove the explicit dependency on MicroProfile Config and enable much simpler testing if, instead, you make this a `Function<? super String, ? extends Optional<String>>`. (Then later instead of `config.getOptionalValue(x, String.class)` you do `config.apply(x)`.)